packages/Telephony/Actions/BatchDTMF.cs

```diff
@@ -0,0 +1,109 @@
+
+        /// <summary>
+        /// Gets or sets the property to assign the result to.
+        /// </summary>
+        [JsonProperty("property")]
+        public StringExpression Property { get; set; }
+
+        /// <inheritdoc/>
+        public override async Task<DialogTurnResult> BeginDialogAsync(DialogContext dc, object options = null, CancellationToken cancellationToken = default(CancellationToken))
+        {
+            //Get value of temrination string from expression
+            string regexPattern = this.TerminationConditionRegexPattern?.GetValue(dc.State);
+
+            if (dc.Context.Activity.Type != "message")
+            {
+                return new DialogTurnResult(DialogTurnStatus.Waiting);
+            }
+
+            //Is the current message the termination string?
+            if (new Regex(regexPattern).IsMatch(dc.Context.Activity.Text))
+            {
+                //If so, store the message in the property specified and end this dialog
+                //Get property from expression
+                string property = this.Property?.GetValue(dc.State);
+
+                dc.State.SetValue(property, dc.Context.Activity.Text);
+                return await dc.EndDialogAsync(cancellationToken: cancellationToken).ConfigureAwait(false);
+            }
+            else
+            {
+                //else, append the message to an aggregation memory state and end the turn
+                dc.State.SetValue(AggregationDialogMemory, dc.Context.Activity.Text);
+                return new DialogTurnResult(DialogTurnStatus.Waiting);
+            }
+        }
+
+        /// <inheritdoc/>
+        public override async Task<DialogTurnResult> ContinueDialogAsync(DialogContext dc, CancellationToken cancellationToken = default)
+        {
+            //Get value of temrination string from expression
+            string regexPattern = this.TerminationConditionRegexPattern?.GetValue(dc.State);
+
+            //append the message to the aggregation memory state
+            var existingAggregation = dc.State.GetValue(AggregationDialogMemory, () => string.Empty);
+            existingAggregation += dc.Context.Activity.Text;
+
+            //Is the current aggregated message the termination string?
+            if (new Regex(regexPattern).IsMatch(existingAggregation))
+            {
+                //If so, save it to the output property and end the dialog
+                //Get property from expression
+                string property = this.Property?.GetValue(dc.State);
+
+                dc.State.SetValue(property, existingAggregation);
+                return await dc.EndDialogAsync(cancellationToken: cancellationToken).ConfigureAwait(false);
+            }
+            else
+            {
+                //else, save the updated aggregation and end the turn
+                dc.State.SetValue(AggregationDialogMemory, existingAggregation);
+                return new DialogTurnResult(DialogTurnStatus.Waiting);
```

Comment:
If my regex is '[1-3]*' and I press the 4 key endlessly, wouldn't this end in an infinite loop?